### PR TITLE
Added #553

### PR DESCRIPTION
**Pull request by user-promi**

### CODE REVIEW

**Comment by Sangi0** [comment]
https://watch.screencastify.com/v/xn54P4yMFjtkmE9EIDfc

File: api/class-mvx-rest-controller.php [hunk 1]
```diff
@@ -1592,6 +1606,40 @@ public function mvx_list_of_work_board_content() {
             }
         }
 
+        /**
+         * pending vendor deactivation request list
+         */
+        if (isset($this->mvx_list_of_request_profile_deletion_vendor()->data) && is_array($this->mvx_list_of_request_profile_deletion_vendor()->data) && !empty($this->mvx_list_of_request_profile_deletion_vendor()->data)) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking invaild . check everywhere

File: api/class-mvx-rest-controller.php [hunk 2]
```diff
@@ -1736,7 +1790,81 @@ public function mvx_task_board_icons_triggers($request) {
...
+            }
+        } elseif (!empty($key) && $key == 'product_assigned_admin') {
+            $users = get_users(array( 'role__in' => array('administrator')));
+            if (!empty($value) && is_array($value)) {
+                foreach ($value as $vendor_id) {
+                    $args = array( 'posts_per_page' => -1, 'author__in' => $vendor_id, 'post_type' => 'product', 'post_status' => 'publish' );
+                    $vendor_query = get_posts($args);
+                    if ($vendor_query) {
+                        foreach ($vendor_query as $product_id){
+                            $update_post_author = array( 'ID' => $product_id->ID, 'post_author' => $users[0]->ID );
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking

File: api/class-mvx-rest-controller.php [hunk 3]
```diff
@@ -2640,11 +2768,45 @@ public function mvx_bulk_todo_pending_product($request) {
                     }
                 }
             }
-        }
+        } elseif (!empty($type) && $type == "requested_profile_deletion") {
+            if (isset($this->mvx_list_of_request_profile_deletion_vendor()->data) && is_array($this->mvx_list_of_request_profile_deletion_vendor()->data) && !empty($this->mvx_list_of_request_profile_deletion_vendor()->data)) {
+                foreach ($this->mvx_list_of_request_profile_deletion_vendor()->data as $key => $value) {
+                    if (is_array($data_list) && !empty($data_list) && isset($data_list[$key])) {
+                        $get_request_profile_deletion_vendor_list[] = $value['id'];
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2787]: $value['id'] checking

File: api/class-mvx-rest-controller.php [hunk 4]
```diff
@@ -3011,6 +3173,39 @@ public function mvx_list_of_pending_question($request, $extra_status = '') {
...
+                    'key'     => '_deactivate_reason',
+                    'value'   => '',
+                    'compare' => '!='
+                ),
+            )
+        );
+        $get_request_profile_deletion_vendors = get_users($args);
+        if (!empty($get_request_profile_deletion_vendors)) {
+            foreach ($get_request_profile_deletion_vendors as $request_profile_deletion_vendor) {
+                $reason = get_user_meta($request_profile_deletion_vendor->ID, '_deactivate_reason', true) ? get_user_meta($request_profile_deletion_vendor->ID, '_deactivate_reason', true) : '';
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: $request_profile_deletion_vendor checking

**Review by moumitahalder — CHANGES_REQUESTED** [review]

**Comment by moumitahalder** [comment]
please add tested

**Pull request merged**